lib/moment-strftime.js

```diff
@@ -35,16 +35,38 @@
   };
 
   moment.fn.strftime = function (format) {
-    var momentFormat, value;
-
-    momentFormat = format;
-
-    Object.keys(replacements).forEach(function (key) {
-      value = replacements[key];
-      momentFormat = momentFormat.replace("%" + key, value);
-    });
-
-    return this.format(momentFormat);
+	// Extract strftime tokens
+	var pattern = /\%([a-z|A-Z]+)/g;
+	var formatTokens = format.match(pattern);
+	
+	if(formatTokens != null){
+		var parse = [], momentString = "";
+		
+		// Build a string to format
+		for (var i in formatTokens){
```

Comment:
I'd like to avoid using `in` (e.g. `foo in bars`) given that it's unreliable when used without `hasOwnProperty` (see, for example, [this writeup](http://brianflove.com/2013/09/05/javascripts-hasownproperty-method/)).  This is why `Object.keys` and `forEach` were being used before.